Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -209,5 +216,62 @@ public class PythonAnalysisSettings {
             }
             public PythonSettings python;
         }
+
+        private sealed class CustomCancellationStrategy : ICancellationStrategy {
+            private readonly string _folderName;
+            private readonly JsonRpc _jsonRpc;
+            private readonly ICancellationStrategy _cancellationStrategy;
+
+            private readonly string _cancellationFolderPath;
+
+            public CustomCancellationStrategy(string folderName, JsonRpc jsonRpc) {
+                this._folderName = folderName ?? throw new ArgumentNullException(nameof(folderName));
+                this._jsonRpc = jsonRpc ?? throw new ArgumentNullException(nameof(jsonRpc));
+
+                this._cancellationFolderPath = Path.Combine(Path.GetTempPath(), "python-languageserver-cancellation", this._folderName);
+
+                this._cancellationStrategy = this._jsonRpc.CancellationStrategy;
+                this._jsonRpc.Disconnected += OnDisconnected;
+
+                try {
+                    Directory.CreateDirectory(this._cancellationFolderPath);
+                } catch { 
```

Comment:
please change these catch all to use `IsCriticalException` extension method like it is done at https://github.com/microsoft/PTVS/blob/63992ffc2d5e8453df0af4896ad3128275ec38f8/Python/Product/PythonTools/PythonTools/PythonFeedbackDiagnosticFileProvider.cs#L74

Reply:
thank you. made not catch critical exceptions.